src/engine/Server.cpp

```diff
@@ -811,11 +811,16 @@ boost::asio::awaitable<void> Server::processQuery(
   //  optional<errorMsg> and optional<metadata> and does this logic
   if (exceptionErrorMsg) {
     LOG(ERROR) << exceptionErrorMsg.value() << std::endl;
-    auto errorResponseJson = composeErrorResponseJson(
-        query, exceptionErrorMsg.value(), requestTimer, metadata);
     if (metadata.has_value()) {
-      LOG(ERROR) << metadata.value().coloredError() << std::endl;
+      try {
```

Comment:
```suggestion
     // The `coloredError()` message might fail because of the different Unicode handling of QLever and ANTLR. Make sure to detect this case so that we can fix it if it happens.
      try {
```